cunumeric/module.py

```diff
@@ -3299,16 +3428,20 @@ def einsum(
             out_modes,
             a,
             b,
-            out=(out if len(operands) == 0 else None),
+            out=(out if len(computed_operands) == 0 else None),
             casting=casting,
             dtype=dtype,
         )
-        operands.append(sub_result)
-    assert len(operands) == 1
-    return operands[0]
+        computed_operands.append(sub_result)
+    assert len(computed_operands) == 1
+    return computed_operands[0]
 
 
-def einsum_path(expr, *operands, optimize="greedy"):
+def einsum_path(
+    expr: str,
+    *operands: ndarray,
+    optimize: Union[bool, list[Any], tuple[Any, ...], str] = "greedy",
+) -> tuple[list[Union[str, int]], str]:
```

Comment:
BTW is there any chance to re-consider the return type here? "tuple where the first element is a specific string literal, followed other things" is a pretty awkward type. A dataclass or TypedDict could be lightweight alternatives.

Reply:
I agree that the return type is weird. (@manopapad can comment if there's a reason this code should stay as it is.) I'm fine with refactoring this function to make the return type more sensible.